collect_app/src/main/java/org/odk/collect/android/utilities/FormUtils.java

```diff
@@ -28,17 +28,13 @@ private FormUtils() {
      * directory with name matching the name of the directory represented by {@code formMediaDir}.
      * <p>
      * E.g. if /foo/bar/baz is passed in as {@code formMediaDir}, jr:// URIs will be resolved to
-     * /odk/root/forms/baz.
+     * projectRoot/forms/baz.
      */
     public static void setupReferenceManagerForForm(ReferenceManager referenceManager, File formMediaDir) {
-        // Clear mappings to the media dir for the previous form that was configured
-        referenceManager.clearSession();
+        referenceManager.reset();
 
-        // This should get moved to the Application Class
-        if (referenceManager.getFactories().length == 0) {
```

Comment:
When I was in here recently (fixing the `StoragePathProvider` call) I didn't think the check made sense at all - why would we care if there are already configured factories? I'm guessing our thinking lines up here, but I just wanted to check.

Reply:
It used to work out because there was only one factory to add -- the one on the ODK root. I think at that point without the check there would just have been lots of factories with the same root. Now we use a whole different context per project so I thought it was clearer to just always reset and configure a fresh factory (as opposed to detecting whether we've changed projects and only reconfiguring if we have).

There's a lot of weirdness here in general. I don't know why JR decided to have different prefixes for different file types. This whole code here is set up to erase that distinction -- we basically say no matter what the prefix is, first pretend it's jr://file and then resolve that.

Reply:
Oh and just another note since I was also confused by all this. There used to be a call on `referenceManager.clearSession()` which did NOT reset the factories. That's why the test had value and why without it there'd be lots of duplicate factories.